galaxy_ng/app/api/ui/viewsets/my_namespace.py

```diff
@@ -0,0 +1,18 @@
+
+from galaxy_ng.app import models
+from galaxy_ng.app.api import permissions
+from galaxy_ng.app.api.v3.viewsets.namespace import NamespaceViewSet
+
+
+class MyNamespaceViewSet(NamespaceViewSet):
+    def get_queryset(self):
+        # All namespaces for users in the partner-engineers groups
+
+        if permissions.IsPartnerEngineer().has_permission(self.request, self):
+            queryset = models.Namespace.objects.all()
+            return queryset
+
+        # Just the namespaces with groups the user is in
+        queryset = models.Namespace.objects.filter(
+            groups__in=self.request.user.groups.all())
+        return queryset
```

Comment:
```suggestion
    def get_queryset(self):
        """
        Returns all namespaces for users in the partner-engineers group, 
        otherwise returns namespaces with groups the user belongs to.
        """
        if permissions.IsPartnerEngineer().has_permission(self.request, self):
            return models.Namespace.objects.all()
        else:
            return models.Namespace.objects.filter(
                groups__in=self.request.user.groups.all()
            )
```